packages/IntegrationTest/windows/integrationtest/TestCommandListener.cpp

```diff
@@ -0,0 +1,153 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include "pch.h"
+#include "TestCommandListener.h"
+
+#include <winrt/Windows.Storage.Streams.h>
+
+using namespace winrt::integrationtest;
+using namespace winrt::Microsoft::ReactNative;
+using namespace winrt::Windows::Data::Json;
+using namespace winrt::Windows::Foundation;
+using namespace winrt::Windows::Networking;
+using namespace winrt::Windows::Networking::Sockets;
+using namespace winrt::Windows::Storage::Streams;
+
+namespace IntegrationTest {
+
+IAsyncAction TestCommandResponse::TestPassed(bool passed) noexcept {
+  JsonObject responseJson;
+  responseJson.SetNamedValue(L"status", JsonValue::CreateStringValue(L"okay"));
+  responseJson.SetNamedValue(L"passed", JsonValue::CreateBooleanValue(passed));
+  co_await SendJson(responseJson);
+}
+
+IAsyncAction TestCommandResponse::Exception(const IRedBoxErrorInfo &err) noexcept {
+  JsonObject responseJson;
+  responseJson.SetNamedValue(L"status", JsonValue::CreateStringValue(L"exception"));
+  responseJson.SetNamedValue(L"message", JsonValue::CreateStringValue(err.Message()));
+  responseJson.SetNamedValue(L"originalMessage", JsonValue::CreateStringValue(err.OriginalMessage()));
+  responseJson.SetNamedValue(L"name", JsonValue::CreateStringValue(err.Name()));
+  responseJson.SetNamedValue(L"componentStack", JsonValue::CreateStringValue(err.ComponentStack()));
+
+  JsonArray callstack;
+
+  for (const auto &frame : err.Callstack()) {
+    JsonObject frameJson;
+    frameJson.SetNamedValue(L"file", JsonValue::CreateStringValue(frame.File()));
+    frameJson.SetNamedValue(L"method", JsonValue::CreateStringValue(frame.Method()));
+    frameJson.SetNamedValue(L"line", JsonValue::CreateNumberValue(frame.Line()));
+    frameJson.SetNamedValue(L"column", JsonValue::CreateNumberValue(frame.Column()));
+    frameJson.SetNamedValue(L"collapse", JsonValue::CreateBooleanValue(frame.Collapse()));
+
+    callstack.Append(frameJson);
+  }
+
+  responseJson.SetNamedValue(L"callstack", callstack);
+
+  co_await SendJson(responseJson);
+}
+
+IAsyncAction TestCommandResponse::UnknownError(std::string_view err) noexcept {
+  JsonObject responseJson;
+  responseJson.SetNamedValue(L"status", JsonValue::CreateStringValue(L"error"));
+  responseJson.SetNamedValue(L"message", JsonValue::CreateStringValue(winrt::to_hstring(err)));
+  co_await SendJson(responseJson);
+}
+
+IAsyncAction TestCommandResponse::SendJson(const JsonObject &payload) noexcept {
```

Comment:
Doesn't currently handle the case where a TestRunner disconnects while receiving a result. We don;t really care if this happens, so I should just eat the socket error that happens if we fail.